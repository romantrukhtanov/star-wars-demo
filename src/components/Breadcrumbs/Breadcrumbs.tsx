import { Breadcrumbs as MUIBreadcrumbs, Link as MUILink, Typography } from '@mui/material';
import { Link } from 'react-router-dom';

export type Breadcrumb = {
  title: string,
  link: string,
  disabled?: boolean,
};

type Props = {
  data: Array<Breadcrumb>
};

export function Breadcrumbs({ data }: Props) {
  return (
    <MUIBreadcrumbs aria-label="breadcrumbs">
      {data.map(renderBreadcrumb)}
    </MUIBreadcrumbs>
  );

  function renderBreadcrumb(item: Breadcrumb) {
    if (item.disabled) {
      return <Typography key={item.title} color="text.primary">{item.title}</Typography>;
    }

    return (
      <MUILink key={item.title} component={Link} underline="hover" color="inherit" to={item.link}>
        {item.title}
      </MUILink>
    );
  }
}
