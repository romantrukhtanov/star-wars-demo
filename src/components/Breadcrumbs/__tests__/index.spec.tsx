import { describe, expect, it } from 'vitest';

import { render } from '@/tests/test-utils';

import type { Breadcrumb } from '../Breadcrumbs';
import { Breadcrumbs } from '../Breadcrumbs';

const mockData: Array<Breadcrumb> = [{
  title: 'Characters',
  link: '/characters',
  disabled: false,
}, {
  title: 'Luke',
  link: '#',
  disabled: true,
}];

describe('Breadcrumbs', () => {
  it('render component', () => {
    const { baseElement } = render(<Breadcrumbs data={mockData} />);
    expect(baseElement).toMatchSnapshot();
  });
});
