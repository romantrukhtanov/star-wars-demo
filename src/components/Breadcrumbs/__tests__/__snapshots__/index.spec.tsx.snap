// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumbs > render component 1`] = `
<body>
  <div>
    <nav
      aria-label="breadcrumbs"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-bvyohc-MuiTypography-root-MuiLink-root"
            href="/characters"
          >
            Characters
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
          >
            Luke
          </p>
        </li>
      </ol>
    </nav>
  </div>
</body>
`;
